pkg/plugins/policies/meshtrace/plugin/v1alpha1/plugin.go

```diff
@@ -43,43 +45,45 @@ func (p plugin) MatchedPolicies(dataplane *core_mesh.DataplaneResource, resource
 
 func (p plugin) Apply(rs *xds.ResourceSet, ctx xds_context.Context, proxy *xds.Proxy) error {
 	policies, ok := proxy.Policies.Dynamic[api.MeshTraceType]
-	if !ok {
+	if !ok || len(policies.SingleItemRules.Rules) == 0 {
```

Comment:
Moved this from https://github.com/kumahq/kuma/pull/14148/files#diff-b8341101e67fe413396a21bb95ff3f30a89ae163e20a45225e48c9569bc181daL181